master/internal/trial.go

```diff
@@ -705,6 +706,10 @@ func (t *trial) processCompletedWorkload(ctx *actor.Context, msg workload.Comple
 		reason := *msg.ExitedReason
 		ctx.Log().Infof("exiting trial early from %v with reason %v", msg.Workload, reason)
 
+		if reason == workload.InitInvalidHP {
+			t.TerminateNoCheckpoint = true
```

Comment:
Can you leave a comment on the decl of this field that describes its purpose? Something like "... this flag marks an immediate exit that is unrecoverable"? Maybe be nice to just call it `UnrecoverableExit` or somthing (the more general/accurate what it does, the more likely it will be reused properly if needed).

Reply:
But, actually don't worry about it because I'm going to refactor this shortly.